linode/instanceconfig/schema_resource.go

```diff
@@ -230,24 +230,93 @@ var helpersSchema = map[string]*schema.Schema{
 	},
 }
 
+const (
+	onlyAllowedForVPCMsg  = "This attribute is only allowed for VPC interfaces."
+	onlyAllowedForVLANMsg = "This attribute is only allowed for VLAN interfaces."
+	requiredForVPCMsg     = "This attribute is required for VPC interfaces."
+	requiredForVLANMsg    = "This attribute is required for VLAN interfaces."
+)
+
 var interfaceSchema = map[string]*schema.Schema{
 	"purpose": {
 		Type:        schema.TypeString,
 		Description: "The type of interface.",
 		Required:    true,
 		ValidateDiagFunc: validation.ToDiagFunc(
-			validation.StringInSlice([]string{"public", "vlan"}, true),
+			validation.StringInSlice([]string{"public", "vlan", "vpc"}, true),
 		),
 	},
-
 	"ipam_address": {
```

Comment:
We may also want to reflect these interface changes in the `linode_instances` data source, although that may be out of scope for now.

Reply:
Yea, just realized the same issue. Working on it now.